libglusterfs/src/glusterfs/fd.h

```diff
@@ -75,6 +75,8 @@ typedef struct _fdtable fdtable_t;
  */
 #define GF_FDENTRY_ALLOCATED -2
 
+#define fd_is_anonymous(_fd) (((_fd) && (_fd)->anonymous))
```

Comment:
Making it a macro means that `_fd` will be evaluated twice. It could just mean more dereferences if compiler don't optimize them, or even duplicated function calls, which is incorrect most probably.

I would use a `static inline` function that will be optimized in the same way by the compiler and doesn't have the drawbacks of a macro.

Reply:
Hmm.. how did that change get in? It's unrelated, probably I'll revert it.